labels/labels.go

```diff
@@ -31,3 +31,35 @@ func ValidateAPI(label string) (string, error) {
 
 	return label, nil
 }
+
+// LabelVersion versions our labels so they can be easily update to contain
+// new data while still easily string matched.
+type LabelVersion uint8
+
+// LabelVersionZero is the label version for labels that contain label type and
+// channel ID (where available).
+const LabelVersionZero LabelVersion = iota
+
+// LabelType indicates the type of label we are creating. It is a string rather
+// than an int for easy string matching and human-readability.
+type LabelType string
+
+var (
+	// LabelTypeChannelOpen is used to label channel opens.
+	LabelTypeChannelOpen LabelType = "openchannel"
+
+	// LabelTypeChannelClose is used to label channel closes.
+	LabelTypeChannelClose LabelType = "closechannel"
+
+	// LabelTypeJusticeTransaction is used to label justice transactions.
+	LabelTypeJusticeTransaction LabelType = "justicetx"
+
+	// LabelTypeSweepTransaction is used to label sweeps.
+	LabelTypeSweepTransaction LabelType = "sweep"
+)
+
+// MakeLabel creates a label with the provided type and channel ID integer
+// representation.
+func MakeLabel(labelType LabelType, channelID uint64) string {
+	return fmt.Sprintf("%v:%v:%v", LabelVersionZero, labelType, channelID)
```

Comment:
Not sure if this format's been used somewhere else before (faraday?) or if we can still change it.
Looks a bit strange to me, for example: `0:openchannel:0`. What about something like `v0:openchannel:chanid-0`?

Also, what do you think about making `channelID` a pointer so it can be nil and logged as such (or as `<unknown>`) instead of just 0?

Reply:
Not locked into anything yet, great suggestion! I've gone with leaving it off completely, thinking of the data fields in the label (ie everything except the version and type) kind of like tlv, as we extend we can {field}-{value} combinations as they are available.